### ISSUE TITLE
Provide better Error Validation messages.

### ISSUE THREAD

**Issue by NikolasKomonen** [angelozerr/lsp4xml#181]
Improve the clarity of error validation messages.

**Comment by angelozerr** [comment]
What do you mean with `clarity`? Error message comes from Xerces, you wish to override it?

**Comment by NikolasKomonen** [comment]
@fbricon yes, for example

```
<project xmlns="http://maven.apache.org/POM/4.0.0" 
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	...
        <a></a>
        ...
```
Gives the output:

[xml] cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://maven.apache.org/POM/4.0.0":a}'. One of '{"http://maven.apache.org/POM/4.0.0":parent, "http://maven.apache.org/POM/4.0.0":name, "http://maven.apache.org/POM/4.0.0":description, "http://maven.apache.org/POM/4.0.0":url, "http://maven.apache.org/POM/4.0.0":inceptionYear, "http://maven.apache.org/POM/4.0.0":organization, "http://maven.apache.org/POM/4.0.0":licenses, "http://maven.apache.org/POM/4.0.0":developers, "http://maven.apache.org/POM/4.0.0":contributors, "http://maven.apache.org/POM/4.0.0":mailingLists, "http://maven.apache.org/POM/4.0.0":prerequisites, "http://maven.apache.org/POM/4.0.0":modules, "http://maven.apache.org/POM/4.0.0":scm, "http://maven.apache.org/POM/4.0.0":issueManagement, "http://maven.apache.org/POM/4.0.0":ciManagement, "http://maven.apache.org/POM/4.0.0":distributionManagement, "http://maven.apache.org/POM/4.0.0":properties, "http://maven.apache.org/POM/4.0.0":dependencyManagement, "http://maven.apache.org/POM/4.0.0":dependencies, "http://maven.apache.org/POM/4.0.0":repositories, "http://maven.apache.org/POM/4.0.0":pluginRepositories, "http://maven.apache.org/POM/4.0.0":build, "http://maven.apache.org/POM/4.0.0":reports, "http://maven.apache.org/POM/4.0.0":reporting, "http://maven.apache.org/POM/4.0.0":profiles}' is expected.

which I feel can be simplified

**Comment by angelozerr** [comment]
I understand your idea, but I'm not sure that's a good idea to override messages. Those messages comes from Xerces and you can use google to understand more your problem, because Xerces is used in an so many project like IJ IDE, Eclipse, etc.

@fbricon what do you think about this issue?

**Comment by fbricon** [comment]
From a user standpoint, those messages are not super useful (dare I say horrible?) as they're very hard to parse. 
From a technical standpoint, I understand it can be tricky to override. 
Now let's keep that issue open for now, as we have other stuff to fix first, but anyone having an idea on how to make it right can chime in.

**Comment by akaroml** [comment]
Not familiar with Xerces but can it provide error message in the format of structured data?

**Comment by NikolasKomonen** [comment]
[Here](https://github.com/apache/xerces2-j/blob/trunk/src/org/apache/xerces/impl/msg/XMLSchemaMessages.properties) are where the messages for schemas are stored in Xerces.

**Comment by akaroml** [comment]
> [Here](https://github.com/apache/xerces2-j/blob/trunk/src/org/apache/xerces/impl/msg/XMLSchemaMessages.properties) is the property file where the messages for schemas are stored in Xerces.
> 
> [This](https://github.com/apache/xerces2-j/blob/e5a239b96fd2cff6566a29e7a4a3a4a2bbf9b0d4/src/org/apache/xerces/impl/xs/XSMessageFormatter.java#L68) is where Xerces assigns the resource path.
> 
> [Here](https://github.com/angelozerr/lsp4xml/blob/e1a5c195e73ea0e974233042635e0de6b299afcd/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java#L52) is where lsp4xml assigns Xerces' XSMessageFormatter, which I assume we can override to point to a modified property file with better messages.
> 
> @angelozerr correct me if I'm wrong.

This is so specific. Thank you for the info.

I have to echo this issue since it's so painful to look at the error info in VS Code. We'll be more confident to recommend this extension to our users with this problem addressed.

**Comment by angelozerr** [comment]
> Here is where lsp4xml assigns Xerces' XSMessageFormatter, which I assume we can override to point to a modified property file with better messages.

I had to do that because I lost Xerces messages. I don't know if it's a clean solution.

To be honnest with you, I have no time to work on this issue. @fbricon @NikolasKomonen is it an important issue for you? Have you planned to work on this issue?

**Comment by NikolasKomonen** [comment]
@angelozerr I am able to work on this issue after my part for DTD issues.

**Comment by fbricon** [comment]
There's a ton of possible error messages to override. I suggest we focus on the 5-10 most common ones for starters, see where it goes.

**Comment by fbricon** [comment]
> I have to echo this issue since it's so painful to look at the error info in VS Code. We'll be more confident to recommend this extension to our users with this problem addressed.

@akaroml you'll get the exact same error messages from Eclipse or IntelliJ so users should not be surprised here. Is it ok? Not really, but at least we're on the same playing field. You should also consider the other features brought to the table when considering recommending the Red Hat's vscode-xml extension.

**Comment by NikolasKomonen** [comment]
If anyone finds messages they would like to be simplified please post them here.
